src/macaron/vsa/vsa.py

```diff
@@ -0,0 +1,251 @@
+# field names in the VSA to conform with in-toto naming conventions.
+
+
+class Vsa(TypedDict):
+    """The Macaron Verification Summary Attestation.
+
+    For reference, see:
+    * `SLSA <https://slsa.dev/spec/v1.0/verification_summary>`_.
+    * `in-toto <https://github.com/in-toto/attestation/blob/main/spec/predicates/vsa.md>`_.
+    """
+
+    #: The payload type. Following in-toto, this is always ``application/vnd.in-toto+json``.
+    payloadType: str  # noqa: N815
+
+    #: The payload of the VSA, base64 encoded.
+    payload: str
+
+
+class VsaStatement(TypedDict):
+    """The Statement layer of a Macaron VSA.
+
+    For reference, see:
+    * in-toto Statement layer specification: https://github.com/in-toto/attestation/blob/main/spec/v1/statement.md.
+    """
+
+    _type: str
+    subject: list[dict]
+    predicateType: str  # noqa: N815
+    predicate: VsaPredicate
+
+
+class VsaPredicate(TypedDict):
+    """The 'predicate' field in the Statement layer of a Macaron VSA.
+
+    For reference, see:
+    * in-toto Predicate layer specification:
+    https://github.com/in-toto/attestation/blob/main/spec/v1/predicate.md.
+    * SLSA VSA predicate schema:
+    https://slsa.dev/spec/v1.0/verification_summary#schema.
+    """
+
+    #: Identity of the verifier, as a tool carrying out the verification.
+    verifier: Verifier
+
+    #: The timestamp when the verification occurred.
+    #: The field has the type
+    # `Timestamp <https://github.com/in-toto/attestation/blob/main/spec/v1/field_types.md#Timestamp>`_.
+    timeVerified: str  # noqa: N815
+
+    #: URI that identifies the resource associated with the software component being verified.
+    #: This field has the type
+    #: `ResourceURI <https://github.com/in-toto/attestation/blob/main/spec/v1/field_types.md#ResourceURI>`_.
+    #: Currently, this has the same value as the subject of the VSA, i.e. the PURL of
+    #: the software component being verified against.
+    resourceUri: str  # noqa: N815
+
+    #: The policy that the subject software component was verified against.
+    #: This field has the type
+    #: `ResourceDescriptor <https://github.com/in-toto/attestation/blob/main/spec/v1/resource_descriptor.md>`_.
+    policy: dict[str, Any]
+
+    #: The verification result.
+    verificationResult: VerificationResult  # noqa: N815
+
+    #: According to SLSA, this field "indicates the highest level of each track verified
+    #: for the artifact (and not its dependencies), or ``FAILED`` if policy verification failed".
+    #: We currently leave this list empty.
+    verifiedLevels: list  # noqa: N815
+
+
+class Verifier(TypedDict):
+    """The 'verifier' field within the Macaron VSA predicate field.
+
+    This field provides the identity of the verifier, as well as the versioning details of its components.
+    """
+
+    #: The identity of the verifier as a value of type
+    #:   `TypeURI <https://github.com/in-toto/attestation/blob/main/spec/v1/field_types.md#TypeURI>`_.
```

Comment:
Same comment regarding the type.

Reply:
See 6870a3ab50fc1342a3727cb0f6aa51ecc0933e33.

Reply:
This line still contains `as a value of type`.

Reply:
See d2d8bbf31e8d676035776e73618c64fbc4ab4a88.